webrender/src/display_list_flattener.rs

```diff
@@ -733,12 +748,19 @@ impl<'a> DisplayListFlattener<'a> {
             }
             SpecificDisplayItem::ClipChain(ref info) => {
                 let items = self.get_clip_chain_items(pipeline_id, item.clip_chain_items())
-                                .iter()
-                                .map(|id| self.id_to_index_mapper.get_clip_node_index(*id))
-                                .collect();
-                let parent = info.parent.map(|id|
-                     self.id_to_index_mapper.get_clip_chain_index(&ClipId::ClipChain(id))
-                );
+                    .iter()
+                    .map(|id| self.id_to_index_mapper.get_clip_node_index(*id))
+                    .collect();
+                let parent = match info.parent {
```

Comment:
I think you can `map_or_else` here:

```rust
let parent = info.parent.map_or_else(
    || self.pipeline_clip_chains.get(&pipeline_id).cloned(),
    |id| self.id_to_index_mapper.get_clip_node_index(*id))
);
```

Reply:
Oh, I see that map_or_else won't wok here with parent accepting an `Option`. Perhaps it makes sense to always fall back to `ClipChainIndex(0)`.